src/animation/universalTransition.ts

```diff
@@ -49,45 +56,104 @@ const getUniversalTransitionGlobalStore = makeInner<GlobalStore, ExtensionAPI>()
 interface DiffItem {
     dataGroupId: string
     data: SeriesData
-    dim: DimensionLoose
+    groupId: string
+    childGroupId: string
     divide: UniversalTransitionOption['divideShape']
     dataIndex: number
 }
 interface TransitionSeries {
     dataGroupId: string
     data: SeriesData
     divide: UniversalTransitionOption['divideShape']
-    dim?: DimensionLoose
+    groupIdDim?: DimensionLoose
+}
+
+enum TransitionDirection {
```

Comment:
We avoid using `enum` in our codebase because it's not in the ES standard. We can change it to simpler constants. For example:

```ts
const TRANSITION_NONE = 0
const TRANSITION_P2C = 1
const TRANSITION_C2P = 2
```

The generated code will also be smaller.